htmresearch/frameworks/nlp/imbu.py

```diff
@@ -127,20 +133,25 @@ def _modelFactory(self, modelType, **kwargs):
     kwargs.update(**self._defaultModelFactoryKwargs())
 
     if modelType == "CioWordFingerprint":
-      kwargs.update(retina=kwargs.get("retina") or self.defaultRetina,
-                    apiKey=kwargs.get("apiKey"),
+      kwargs.update(modelDir=savePath,
+                    retina=self.retina,
+                    apiKey=self.apiKey,
```

Comment:
It's only considered for types that interact with cortical.io, I.e. these three. Iirc, not all types do, and there may be more in the future.  If we make it required for all models, then yes, we can move it back into `_defaultModelFactoryKwargs()` as it was before, but I don't think that's the right pattern moving forward.